[system] Add empty interfaces to fix issue with typescript module augmentation

yonatan0 opened the pull request:
fix for #43855 to allow typescript module augmentation

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/material-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

Janpot commented on packages/mui-system/src/createTheme/createTheme.d.ts:
```diff
@@ -34,12 +39,12 @@ export interface Theme extends CssContainerQueries {
-  typography?: unknown;
+  mixins?: Mixins;
+  typography?: Typography;
   zIndex?: unknown;
```
The only `unknown` left is `zIndex`. Is there a reason not to type it the same way as `ThemeOptions`?

yonatan0 replied:
my bad, fixed 🙏

yonatan0 commented:
1. Changed the zIndex type from Record<string, number> to a more generic interface: The original type was causing conflicts with @mui/material's createTheme function.
2. Renamed zIndex to ZIndex: to comply with the linting rules that enforce PascalCase naming conventions.

yonatan0 commented:
hey @Janpot,
i tried to figure out why the `argos` ci step failed, but with no success, maybe you have some insights for me on how to fix this?

Janpot reviewed (approved):
This seems good to me

> maybe you have some insights for me on how to fix this?

This screenshot is flaky, we haven't fully figured out why yet. You can safely ignore

mnajdova reviewed (approved):
It's a great first conutrion on MUI 👌 People will appreciate the fix! :)
